data-processing-lib/ray/src/data_processing/transform/abstract_transform.py

```diff
@@ -0,0 +1,16 @@
+from typing import Any, Generic, TypeVar
+
+
+DATA = TypeVar("DATA")
+
+
+class AbstractTransform(Generic[DATA]):
+    def transform(self, data: DATA) -> tuple[list[DATA], dict[str, Any]]:
+        """
+        Converts input table into an output table.
+        If there is an error, an exception must be raised - exit()ing is not generally allowed when running in Ray.
+        :param table: input table
+        :return: a tuple of a list of 0 or more converted tables and a dictionary of statistics that will be
+        propagated to metadata
+        """
+        raise NotImplemented()
```

Comment:
It formalizes the notion of a "transform" which we can then use across runtimes.  I'd like to have some commonality across them and not look like we're reinventing.

Reply:
This is a very noble intention, but do we realy need this? Can you name at least one realistic use case where this can be used? At the moment, its an extra baggage that does not add anything